tests/engine/adhoc_test.go

```diff
@@ -221,3 +225,41 @@ func importedAndExportedFunc(t *testing.T, newEngine func() wasm.Engine) {
 	require.NoError(t, err)
 	require.Equal(t, uint64(42), results[0])
 }
+
+// Test that host functions can handle float parameter without corrupting the value
+func hostFuncWithFloat32Param(t *testing.T, newEngine func() wasm.Engine) {
+	ctx := context.Background()
+	mod, err := text.DecodeModule([]byte(`(module
+		;; 'test_f32param' is a host function which accepts a f32 param
+		;; and tests if it is an expected value.
+		(import "env" "test_f32param" (func $test_f32param (param f32)))
+
+		;; 'call_test_f32' is an exported guest function to call
+		;; the host function, 'test_f32param', from the guest.
+		(func $call_test_f32param (param f32)
+			local.get 0
+			call $test_f32param
+		)
+		(export "call_test_f32param" (func $call_test_f32param))
+		)`))
+	require.NoError(t, err)
+
+	store := wasm.NewStore(newEngine())
+
+	expectedFloat32Val := float32(math.MaxFloat32) // arbitrary float32 value.
+	// testF32Param is called from the guest function, call_test_f32param.
+	// This function expects the guest to pass expectedF32Val.
+	testF32Param := func(ctx *wasm.HostFunctionCallContext, value float32) {
+		require.Equal(t, expectedFloat32Val, value)
+	}
+	err = store.AddHostFunction("env", "test_f32param", reflect.ValueOf(testF32Param))
+	require.NoError(t, err)
+
+	err = store.Instantiate(mod, "test")
+	require.NoError(t, err)
+
+	// `call_test_f32param` calls the host function `testF32Param` with `expectedF32Val` as its parameter,
```

Comment:
In general, we don't need to summarize the signature if it is one line below. The impact is the important part. Ex. 

`This invokes call->test.f32param to ensure float parameters aren't corrupted when routing through OpCodeCall`

Reply:
Updated as suggested. Thanks.